### PR TITLE
Upload test

### PR DESCRIPTION
**Pull request by scholtz-gnome**
This test for file upload "works", but it feels a little hacky. Would appreciate your eyes on this. I've included another solution commented out, but that wasn't as clean.

### CODE REVIEW

File: tests/songController.test.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import request from "supertest";
+import { newApp } from "../src/app";
+import fs from "fs";
+
+const app = newApp();
+
+describe("Upload a pdf", () => {
```
Comment (jcqvisser) [inline tests/songController.test.ts:7]: Remember to wrap async functions in async functions, else the app may stop executing before the inside `async` functions complete.

```suggestion
describe("Upload a pdf", async () => {
```

**Review by jcqvisser — CHANGES_REQUESTED** [review]

File: tests/songController.test.ts (hunk 1 above)
Comment (scholtz-gnome) [reply tests/songController.test.ts:7]: I did not know that! Thank you!

File: tests/songController.test.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  const testFilePath = `${__dirname}/testUploads/test.pdf`;
```
[the remark this answers is not in the document]
Comment (jcqvisser) [reply tests/songController.test.ts]: `promisify` is cool. It does something like this (in JS and simplified and also pulled out of my ass):

```js
const promisify = (fn) => {
  return (arg) => {
    return new Promise((resolve, reject) => {
      fn(
        arg,
        (successValue) => { resolve(successValue); },
        (failureValue) => { reject(failureValue); }
      );
    });
  }
};
```

[the remark this answers is not in the document]
Comment (scholtz-gnome) [reply tests/songController.test.ts]: Cool! I suppose it's aptly named then.

**Comment by scholtz-gnome** [comment]
Closing PR for new changes in debugging-fix

**Pull request closed without merging**